autocorrect/src/rule/rule.rs

```diff
@@ -35,27 +37,27 @@ impl Rule {
             return;
         }
 
-        let new = (self.format_fn)(&result.out);
-        if result.out.ne(&new) {
+        if let Cow::Owned(new) = (self.format_fn)(&result.out) {
             result.severity = Severity::Error;
+            result.out = Cow::Owned(new);
         }
-        result.out = new;
     }
 
     pub fn lint(&self, result: &mut RuleResult) {
         if self.severity() == SeverityMode::Off {
             return;
         }
 
-        let new = (self.format_fn)(&result.out);
-        if result.out.ne(&new) && result.severity == Severity::Pass {
-            if self.severity() == SeverityMode::Warning {
-                result.severity = Severity::Warning;
-            } else {
-                result.severity = Severity::Error;
+        if let Cow::Owned(new) = (self.format_fn)(&result.out) {
+            if result.severity == Severity::Pass {
```

Comment:
Missed changed check here.

Reply:
This is inside the `Cow::Owned` block, which means it's changed already.